### PR TITLE
(feat) Add v0.1.2 for dkp-insights

### PR DESCRIPTION
**Pull request by kaiwalyajoshi**
- Uses v0.1.2 of `dkp-insights`. 
- This is app for the attached clusters and is sourced from `http://mesosphere.github.io/dkp-insights-charts-attached`

### CODE REVIEW

File: services/dkp-insights/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
...MDE5LjQgMCAuODUzIDAgMS41NDgtLjY4OSAxLjU1OS0xLjU0NXYtLjI2MWwyNC41MDUtNi41ODJ6bS0yNi43NDMgOC4yODdjLS43NCAyLjMwOC0xLjk5OSA2LjY2Mi0yLjUzOCA4LjgwOWgtLjE2Yy0uODUzLjAxMS0xLjUzOS43MDktMS41MzkgMS41NjVhMS41OSAxLjU5IDAgMCAwIC4yLjc0Mkw4Ny41NCA2MC41MzhsMjEuNDQ2LTI4LjQ1NC4wOC4wNnptLTEyLjI1MiA5Mi4zMDZoMGwtOS45OTQtMTguOTAzIDIyLjQ2NiAzMC40NjFoMGwtMTEuMzczLTEwLjI1NHYtLjE4MWExLjE0IDEuMTQgMCAwIDAtMS4wOTktMS4xMjRoMHptMjguNjAyIDIxLjg3MmExLjE2IDEuMTYgMCAwIDAtLjk1OS41MjJsLTEyLjc5Mi04LjgyOSAzNS45NzcgMTIuODQzLTIxLjEyNy0zLjE3Yy4wMDktLjA3My4wMDktLjE0NyAwLS4yMjFhMS4xNCAxLjE0IDAgMCAwLTEuMDk5LTEuMTQ0aDB6bTM1LjM1OCAxLjM2NWMtLjE4OC0uMTQ3LS40MjEtLjIyNS0uNjYtLjIyMWExLjE2IDEuMTYgMCAwIDAtMS4wNTkuNzIybC04LjM3NSAyLjUwOGMtLjAwNi0uMDUzLS4wMDYtLjEwNyAwLS4xNjFsMjAuNjQ3LTE0LjYwOGMuMzY0LjQwNS44NzYuNjQ0IDEuNDE5LjY2MmExLjg0IDEuODQgMCAwIDAgMS42OTktMS4xMjRsMTIuMTEyIDQuMDEzYy0uMDEuMTE0LS4wMS4yMjggMCAuMzQxdi4yODFsLTI1Ljc4NCA3LjU4NXptMjcuOTgyLTkuMjUxaDBsMTEuMTMzLTI1LjA0M2ExLjk5IDEuOTkgMCAwIDAgLjYyLjEyYzEuMDExLS4wMTEgMS44MjgtLjgzMSAxLjgzOS0xLjg0NmEyLjUyIDIuNTIgMCAwIDAgMC0uNDIxbDcuOTk1LTIuNDA4Yy4wNzcuMjExLjIwOC4zOTguMzguNTQybC0yMS45NjYgMjkuMDU2eiIgc3Ryb2tlPSIjMzEwMDVkIiBzdHJva2Utd2lkdGg9Ii4xNiIgZmlsbD0iIzMxMDA1ZCIvPjwvZz48L2c+PC9zdmc+
```
Comment (kaiwalyajoshi) [inline services/dkp-insights/metadata.yaml:12]: @nataliepina could I get the icon for Insights from you, I'm not sure where to find it. I think this is a png image.

**Comment by kaiwalyajoshi** [comment]
@takirala @asekretenko

This is the first pass for landing this.

I followed these examples:
- [spark-operator-1.17 catalog-app](https://github.com/mesosphere/dkp-catalog-applications/tree/main/services/spark-operator/1.1.17)
- [kaptain catalog-applications](https://github.com/mesosphere/kaptain-catalog-applications)

The way that the chart is specified in `services/dkp-insights/0.10.0/dkp-insights.yaml` follows spark-operator, I'm not sure if this is correct and could use guidance.

File: services/dkp-insights/0.10.0/defaults/cm.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+metadata:
+  name: dkp-insights-0.10.0-d2iq-defaults
+  namespace: ${releaseNamespace}
+data:
+  values.yaml: |
+    image:
+      registry: "docker.io"
+      repository: "mesosphere/insights"
+      tag: "v0.1.0-rc1"
+      imagePullPolicy: "Always"
```
Comment (asekretenko) [inline services/dkp-insights/0.10.0/defaults/cm.yaml:13]: Can/should we switch to 'IfNotPresent"?

File: services/dkp-insights/0.10.0/defaults/cm.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    backend:
+       log_level: "DEBUG"
```
Comment (asekretenko) [inline services/dkp-insights/0.10.0/defaults/cm.yaml:15]: Wondering what level we should use by default. 
Do we want to ask users to switch from DEBUG to INFO if the logging volume becomes too high, or to switch from INFO to DEBUG if something goes wrong?

**Review by asekretenko — APPROVED** [review]
LGTM; raised a couple of questions regarding parameters which we might want to set differently from the dev environment.

File: helm-repositories/dkp-insights/dkp-insights-repo.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: HelmRepository
+metadata:
+  name: mesosphere.github.io-dkp-insights-charts-attached
+  namespace: ${workspaceNamespace}
```
Comment (takirala) [inline helm-repositories/dkp-insights/dkp-insights-repo.yaml:6]: Does this need to work in airgapped? If yes, we need https://github.com/mesosphere/dkp-catalog-applications/blob/main/helm-repositories/mesosphere-spark-repo/mesosphere-spark-repo.yaml#L8 (This is needed in addition to the `helmMirrorUrl` that is already added few lines below)

File: services/dkp-insights/0.10.0/defaults/cm.yaml (hunk 2 above)
Comment (takirala) [reply services/dkp-insights/0.10.0/defaults/cm.yaml:13]: +1

**Review by takirala — APPROVED** [review]

**Comment by takirala** [comment]
we can do 
- [build the airgapped image tar](https://github.com/mesosphere/dkp-catalog-applications/pull/56/files )
- [build the helm chart tar](https://github.com/mesosphere/kommander/blob/main/Makefile#L441-L449) and then [upload to s3](https://github.com/mesosphere/kommander/blob/main/make/release.mk#L168)

File: helm-repositories/dkp-insights/dkp-insights-repo.yaml (hunk 4 above)
Comment (kaiwalyajoshi) [reply helm-repositories/dkp-insights/dkp-insights-repo.yaml:6]: Yeah, I saw that. It does need to work in airgapped but we haven't verified it yet. Should we add the label after verification?

File: helm-repositories/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - banzaicloud
```
Comment (kaiwalyajoshi) [inline helm-repositories/kustomization.yaml:4]: @takirala will re-adding `banzaicloud` here conflict with https://github.com/mesosphere/dkp-catalog-applications/blob/c1a441bf03b658a7410072401db332af396941f3/helm-repositories/kustomization.yaml#L4?

File: make/release.mk [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	@$(MINDTHEGAP_BIN) create image-bundle --platform linux/amd64 --images-file $(CATALOG_IMAGES_TXT) --output-file $(IMAGE_TAR_FILE)
+	@ls -latrh $(IMAGE_TAR_FILE)
+
+.PHONY: release.s3
+release.s3:
+	$(call print-target)
+ifeq ($(CATALOG_APPLICATIONS_VERSION),"")
+	$(info CATALOG_APPLICATIONS_VERSION should be set to the version which is part of the s3 file path)
+else
+	aws s3 cp --no-progress --acl bucket-owner-full-control $(IMAGE_TAR_FILE) s3://$(RELEASE_S3_BUCKET)/airgapped/$(CATALOG_APPLICATIONS_VERSION)/dkp_insights_image_bundle_$(CATALOG_APPLICATIONS_VERSION)_linux_amd64.tar.gz
```
Comment (kaiwalyajoshi) [inline make/release.mk:25]: @takirala Please note the upload location here.

We'll be uploading `dkp-insights-image-bundle.tar.gz` at a sub-folder under `CATALOG_APPLICATIONS_VERSION` is this what we want or should we go with some other scheme?

**Comment by kaiwalyajoshi** [comment]
Requesting review from @cbuto as well since he's worked on the tooling for `dkp-catalog-applications'

File: helm-repositories/kustomization.yaml (hunk 5 above)
Comment (cbuto) [reply helm-repositories/kustomization.yaml:4]: I think we'll just have to make sure the HelmRepo is not named the same i.e. https://github.com/mesosphere/dkp-catalog-applications/blob/c1a441bf03b658a7410072401db332af396941f3/helm-repositories/banzaicloud/banzaicloud.yaml#L5

File: make/release.mk (hunk 6 above)
Comment (cbuto) [reply make/release.mk:25]: right now, the dkp-catalog-applications CI that runs this make target is triggered via tags `+:v*`

So we just set `CATALOG_APPLICATIONS_VERSION=%teamcity.build.branch%"` which ends up being the tag value (i.e. v2.2.0).

File: mergebot-config.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    "update_jira_on_merge": {
+      "enabled": true,
+      "interactive": false,
+      "fix_version_map": {
+        "main": "Kommander 2.2.0"
+      },
+      "jira_regex": "((?:COPS|D2IQ)-\\d+)"
+    },
+    "test-binding": {
+      "dkp-catalog-applications/pre-commit": {
```
Comment (cbuto) [inline mergebot-config.json:17]: the references to `dkp-catalog-applications` likely need to change

File: mergebot-config.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        "type": "teamcity",
+        "required": "true",
+        "priority": 2,
+        "id": "ClosedSource_Kommander2_DkpCatalogApplications_PreCommitChecks"
```
Comment (cbuto) [inline mergebot-config.json:21]: This ID should eventually point to the insights TC jobs once they are created

File: helm-repositories/kustomization.yaml (hunk 5 above)
Comment (kaiwalyajoshi) [reply helm-repositories/kustomization.yaml]: okay, I've fixed this.

File: helm-repositories/dkp-insights/dkp-insights-repo.yaml (hunk 4 above)
Comment (takirala) [reply helm-repositories/dkp-insights/dkp-insights-repo.yaml:6]: there is a ticket to update the docs but i agree this is a bit confusing. 
we need this label if this has to work in airgapped (label is invisible to user). to certify that a package works in airgapped (making visible to user), we update `metadata.yaml` with certified label e.g.: https://github.com/mesosphere/dkp-catalog-applications/blob/c1a441bf03b658a7410072401db332af396941f3/services/kafka-operator/metadata.yaml#L7

**Pull request merged**